indico/modules/attachments/templates/_attachments.html

```diff
@@ -8,14 +8,18 @@
 
 {% macro render_attachment(attachment, has_label=false, classes='') %}
     {% set previewable = (attachment.file.is_previewable and not g.static_site) or false %}
+    {% set restricted = attachment.is_self_protected and not attachment.can_access(session.user) %}
```

Comment:
let's not perform the access check unless we need it (ie it's actually a link attachment)

Reply:
I moved these up because it was getting a bit confusing, but that's probably better.

Reply:
that makes sense, I'd just merge this into the `set visible_link` statement so we short-circuit in case it's not a link